tracing-mock/README.md

```diff
@@ -0,0 +1,181 @@
+tracing-core = { git = "https://github.com/tokio-rs/tracing", branch = "v0.1.x" }
+```
+
+## Examples
+
+The following examples are for the `master` branch. For examples that
+will work with `tracing` from [crates.io], please check the
+[v0.1.x](https://github.com/tokio-rs/tracing/tree/v0.1.x/tracing-mock)
+branch.
+
+Below is an example that checks that an event contains a message:
+
+```rust
+use tracing::collect::with_default;
+use tracing_mock::{collector, event, field};
+
+fn yak_shaving() {
+    tracing::info!("preparing to shave yaks");
+}
+
+#[test]
+fn traced_event() {
+    let (collector, handle) = collector::mock()
+        .event(event::mock().with_fields(field::msg("preparing to shave yaks")))
+        .done()
+        .run_with_handle();
+
+    with_default(collector, || {
+        yak_shaving();
+    });
+
+    handle.assert_finished();
+}
+```
+
+Below is a slightly more complex example. `tracing-mock` asserts that, in order:
+- a span is created with a single field/value pair
+- the span is entered
+- an event is created with the field `number_of_yaks`, a corresponding
+  value of 3, and the message "preparing to shave yaks", and nothing else
+- an event is created with the field `all_yaks_shaved`, a corresponding value
+  of `true`, and the message "yak shaving completed"
+- the span is exited
+- the span is closed
+- no further traces are received
+
+```rust
+use tracing::collect::with_default;
+use tracing_mock::{collector, event, field, span};
+
+#[tracing::instrument]
+fn yak_shaving(number_of_yaks: u32) {
+    tracing::info!(number_of_yaks, "preparing to shave yaks");
+
+    let number_shaved = number_of_yaks; // shave_all
+    tracing::info!(
+        all_yaks_shaved = number_shaved == number_of_yaks,
+        "yak shaving completed."
+    );
+}
+
+#[test]
+fn yak_shaving_traced() {
+    let yak_count: u32 = 3;
+    let span = span::mock().named("yak_shaving");
+
+    let (collector, handle) = collector::mock()
+        .new_span(
+            span.clone()
+                .with_field(field::mock("number_of_yaks").with_value(&yak_count).only()),
+        )
+        .enter(span.clone())
+        .event(
+            event::mock().with_fields(
+                field::mock("number_of_yaks")
+                    .with_value(&yak_count)
+                    .and(field::msg("preparing to shave yaks"))
+                    .only(),
+            ),
+        )
+        .event(
+            event::mock().with_fields(
+                field::mock("all_yaks_shaved")
+                    .with_value(&true)
+                    .and(field::msg("yak shaving completed."))
+                    .only(),
+            ),
+        )
+        .exit(span.clone())
+        .close_span(span)
+        .done()
+        .run_with_handle();
+
+    with_default(collector, || {
+        yak_shaving(yak_count);
+    });
+
+    handle.assert_finished();
+}
+```
+
+The full code for both examples can be found in the readme.rs tests.
```

Comment:
since these are now being run as doctests, this is no longer accurate, right?

Reply:
Correct, removed.